azurerm/internal/services/containers/kubernetes_cluster_node_pool_data_source.go

```diff
@@ -0,0 +1,262 @@
+package containers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-03-01/containerservice"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceKubernetesClusterNodePool() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceKubernetesClusterNodePoolRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.KubernetesAgentPoolName,
+			},
+
+			"kubernetes_cluster_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			// Computed
+			"availability_zones": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"enable_auto_scaling": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"enable_node_public_ip": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"eviction_policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"max_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"max_pods": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"mode": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"min_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"node_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"node_labels": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"node_taints": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"orchestrator_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"os_disk_size_gb": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"os_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"priority": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"spot_max_price": {
```

Comment:
This name is inconsistent with `windows_virtual_machine_scale_set` and `linux_virtual_machine_scale_set`

Reply:
unfortunately different teams within Microsoft can refer to the same functionality in different ways - as such unfortunately we need to use the name being exposed by AKS here (which is `spotMaxPrice` - rather than "spot bid price" used by the VM Scale Set API's)